automated_analysis.py

```diff
@@ -464,5 +459,42 @@ def set_survey_percentages(survey_counts, total_survey_counts):
         fig.update_xaxes(tickangle=-60)
         fig.write_image(f"{automated_analysis_output_dir}/graphs/{plan.raw_field}_by_gender_normalised.png",
                         scale=IMG_SCALE_FACTOR)
+    
+    # Export safe to share raw messages for each episode to share with WUSC
+    log.info("Exporting safe to share raw messages for each episode...")
+    safe_to_share_messages = []  # of dict
+    for plan in PipelineConfiguration.RQA_CODING_PLANS[5:]: #loop through episode 6 -> because previous episodes had been manually processed
+        for cc in plan.coding_configurations:
+            code_to_messages = dict()
+            for code in cc.code_scheme.codes:
+                code_to_messages[code.string_value] = []
+
+            for msg in messages:
+                if not AnalysisUtils.opt_in(msg, CONSENT_WITHDRAWN_KEY, plan):
+                    continue
+
+                for label in msg[cc.coded_field]:
+                    code = cc.code_scheme.get_code_with_code_id(label["CodeID"])
+                    code_to_messages[code.string_value].append(msg[plan.raw_field])
+
+            for code_string_value in code_to_messages:
+                if code_string_value == "DNS": # TODO add DNS control code to core and update here
+                    log.warning(f"Excluding {len(code_to_messages[code_string_value])} unsafe to share messages")
+                    continue
+
+                for msg in code_to_messages[code_string_value]:
+                    safe_to_share_messages.append({
+                        "Episode": plan.dataset_name,
+                        "Code": code_string_value,
+                        "Raw Message": msg
+                    })
+
+    with open(f"{automated_analysis_output_dir}/wusc_safe_to_share_messages.csv", "w") as f:
```

Comment:
Is it ok for this to just be 'safe_to_share_messages', so that we don't have to remember to change the prefix between projects? The main case I think of for it not being is if we need to share to multiple organisations with different levels of strictness over what we share, but I don't know how likely we are to ever see that in practice (and in such cases it would probably be easier just to adopt the most strict definition for all those organisations)

Reply:
True, updated to safe_to_share_messages